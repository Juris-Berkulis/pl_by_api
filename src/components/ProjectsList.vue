<script setup lang="ts">
import ProjectsListItem from '@/components/ProjectsListItem.vue';
import { type Project } from '@/types/index';

interface Props {
    projectsList: Array<Project>,
};

defineProps<Props>();
</script>

<template>
<ul>
    <ProjectsListItem v-for="project of projectsList" :key="project.id" :project="project" />
</ul>
</template>

<style scoped>
</style>
