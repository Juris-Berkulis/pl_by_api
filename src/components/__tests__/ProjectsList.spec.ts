import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import ProjectsList from "@/components/ProjectsList.vue";
import type { Project } from "@/types";

describe('ProjectsList', async () => {
    const response: Response = await fetch(`https://dev.aicap.tech/api/v1/interview/projects?page=1`);
    const fetchResult = await response.json();
    const projectsList = fetchResult.results;

    const wrapper = mount(ProjectsList, {
        props: {
            projectsList,
        }
    });

    it('Snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Contains text', () => {
        projectsList.forEach((project: Project) => {
            expect(wrapper.text()).toContain(project.title);
            expect(wrapper.text()).toContain(project.status);
        });
    });
});
