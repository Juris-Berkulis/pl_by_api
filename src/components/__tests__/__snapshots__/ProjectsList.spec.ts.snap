// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectsList > Snapshot 1`] = `
"<ul>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">IN_PROGRESS</p>
    <h2>Рекламный ролик №56 (банки, дебет, 2 кат)</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">IN_PROGRESS</p>
    <h2>Рекламный ролик №55 (банки, вклады, 2 кат)</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">IN_PROGRESS</p>
    <h2>Рекламный ролик №54 (банки, дебет, 2 кат)</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">IN_PROGRESS</p>
    <h2>Рекламный ролик №53 (банки, дебет, 2 кат)</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">IN_PROGRESS</p>
    <h2>Рекламный ролик №52 (банки, дебет, 2 кат)</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">FINISHED</p>
    <h2>Рекламный видеоролик №51 (КАЗ)</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">FINISHED</p>
    <h2>Рекламный видеоролик №50 (РУС)</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">FINISHED</p>
    <h2>Задание от AICAP №8</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">IN_PROGRESS</p>
    <h2>Задание от AICAP №7</h2>
  </li>
  <li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
    <p class=\\"mb-0.5 font-bold\\">IN_PROGRESS</p>
    <h2>Рекламный ролик №48 (банки, дебет, 1 кат)</h2>
  </li>
</ul>"
`;
